geth/node/txqueue_manager.go

```diff
@@ -253,6 +257,54 @@ func (m *TxQueueManager) completeRemoteTransaction(queuedTx *common.QueuedTx, pa
 	return signedTx.Hash(), nil
 }
 
+func (m *TxQueueManager) estimateGas(args common.SendTxArgs) (*hexutil.Big, error) {
+	if args.Gas != nil {
+		return args.Gas, nil
+	}
+
+	client := m.nodeManager.RPCClient()
+	ctx, cancel := context.WithTimeout(context.Background(), time.Minute)
+	defer cancel()
+
+	var gasPrice hexutil.Big
```

Comment:
The logic starting from here until `client.CallContext()` is based on https://github.com/ethereum/go-ethereum/blob/9e5f03b6c487175cc5aa1224e5e12fd573f483a7/eth/bind.go#L81-L97

Reply:
Also, we have similar method in our go-ethereum fork https://github.com/status-im/go-ethereum/blob/status/1.6.7-stable/internal/ethapi/status_backend.go#L64-L88 but only available via `les` service. I preferred not to include it in a method that calls upstream.